Mathlib/Data/Nat/Factorial/NatCast.lean

```diff
@@ -66,4 +66,34 @@ theorem natCast_factorial_iff_of_charP {n : ℕ} : IsUnit (n ! : A) ↔ n < p :=
 
 end CharP
 
+section Nilpotent
+
+variable {A : Type*} [CommRing A] {p : ℕ} [Fact (Nat.Prime p)]
+
+lemma natCast_of_lt_of_isNilpotent (hp : IsNilpotent (p : A)) {n : ℕ} (h₀ : n ≠ 0) (h : n < p) :
```

Comment:
This statement doesn't mention the factorial; perhaps it should be placed elsewhere?

Reply:
I see your point, but I am not sure that there is a simpler file that imports `Nat.Prime`, `IsUnit` and `IsNilpotent`. I tried `#find_home! natCast_of_lt_of_isNilpotent` and this file is the only suggestion.